cleanlab_studio/studio/studio.py

```diff
@@ -266,7 +268,30 @@ def poll_cleanset_status(self, cleanset_id: str, timeout: Optional[int] = None)
         Returns:
             After cleanset is done being generated, returns `True` if cleanset is ready to use, `False` otherwise.
         """
-        return clean_helpers.poll_cleanset_status(self._api_key, cleanset_id, timeout)
+        warnings.warn(
+            "Poll cleanset status method has been deprecated -- please use wait_for_cleanset_ready method instead.",
+            DeprecationWarning,
+        )
+
+        try:
+            clean_helpers.poll_cleanset_status(self._api_key, cleanset_id, timeout)
+            return True
+
+        except (TimeoutError, CleansetRunError):
+            return False
+
+    def wait_until_cleanset_ready(self, cleanset_id: str, timeout: Optional[float] = None) -> None:
+        """Blocks until cleanset is ready, timeout is reached, or cleanset errors.
+
+        Args:
+            cleanset_id (str): ID of cleanset to check status or
+            timeout (Optional[float], optional): timeout for polling, in seconds. Defaults to None.
```

Comment:
I don't remember what syntax we're following here, but why do we have "Optional[float]" as well as "optional"?

Reply:
`Optional` (uppercase `O`) means the value can be `None`.

`optional` (lowercase `o`) means the caller of this function doesn't have to supply a value here (as it defaults to None)